src/ucp/core/ucp_ep.c

```diff
@@ -256,24 +256,25 @@ static void ucp_ep_delete(ucp_ep_h ep)
         ucs_list_del(&ucp_ep_ext_gen(ep)->ep_list);
     }
 
-    if (!(ep->flags & UCP_EP_FLAG_FAILED)) {
-        ucp_ep_release_id(ep);
-    }
-
+    ucp_ep_release_id(ep);
     ucp_ep_destroy_base(ep);
 }
 
 void ucp_ep_release_id(ucp_ep_h ep)
 {
     ucs_status_t status;
 
-    ucs_assert(!(ep->flags & UCP_EP_FLAG_FAILED));
+    if (ucp_ep_ext_control(ep)->local_ep_id == UCP_EP_ID_INVALID) {
```

Comment:
why needed?

Reply:
why would id be invalid?

Reply:
because it already released in UCP error-handling flow

Reply:
https://github.com/openucx/ucx/blob/9efeadc5c7521a44553127b9cb3c89819942cdd7/src/ucp/core/ucp_worker.c#L509

Reply:
so issue is we release ep id twice?
can we release the id same place we set ep to failed? so FAILED flag check will prevent duplicate id release

Reply:
> so issue is we release ep id twice?

yes, we tries to release it twice

>can we release the id same place we set ep to failed? so FAILED flag check will prevent duplicate id release

ok, done